dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -135,4 +135,24 @@ public override string ToString()
     {
         return this.ErrorOccurred ? $"Error: {this.LastErrorDescription}" : this.Result;
     }
+
+    /// <summary>
+    /// Create a clone of the current context, using the same kernel references (memory, skills, logger)
+    /// and a new set variables, so that variables can be modified without affecting the original context.
+    /// </summary>
+    /// <returns>A new context copied from the current one</returns>
+    public SKContext Clone()
```

Comment:
If it's useful, we could implement [IClonable](https://learn.microsoft.com/en-us/dotnet/api/system.icloneable?view=net-7.0) and use [object.MemberwiseClone](https://learn.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-7.0#system-object-memberwiseclone). Same for ContextVariables. <<keep in mind for future change -- don't block this one>>

Reply:
thanks - I took a look, the main drawback I guess is that ICloneable.Clone() would require a cast, e.g. require more code. Once we get into distributed workflows, durable functions etc we'll probably look into these and related features.

Reply:
FWIW, we largely consider `ICloneable` to have been a mistake on our part, and we haven't implemented it on anything new since the early days of .NET.  The primary issue with it is it's underspecified whether it's a shallow or deep clone or somewhere in between, making it difficult to reason about abstractly.